subprojects/core/src/main/java/org/gradle/api/internal/file/copy/DefaultCopySpec.java

```diff
@@ -265,7 +265,7 @@ public DuplicatesStrategy getDuplicatesStrategy() {
     }
 
     @Override
-    public void setDuplicatesStrategy(@Nullable DuplicatesStrategy strategy) {
+    public void setDuplicatesStrategy(DuplicatesStrategy strategy) {
         this.duplicatesStrategy = strategy;
```

Comment:
I guess it would be prudent to check for `null` here?

Reply:
I did consider it and given that:
 * Kotlin won't allow `null` to be given as an argument
 * IntelliJ would warn for Java code passing `null`
 * null checking parameters is not common in the Gradle code base
 * 5.0 is a breaking release

I decided it wasn't worth it but I don't feel strongly about it.

Reply:
We still have Groovy scripts, but yeah. I guess we should have a global decision about how we check for null values.